syntax = "proto3";

package iroha;

service Query {
  rpc GetAccount(AccountRequest) returns (AccountResponse) {}
  rpc GetBalance(BalanceRequest) returns (BalanceResponse) {}
}

message AccountRequest {
  uint64 account_id = 1;
}

message AccountResponse {
  string name = 1;
}

message BalanceRequest {
  uint64 account_id = 1;
  uint64 asset_id = 2;
}

message BalanceResponse {
  uint64 amount = 1;
}
